jazz_create-serverless-service/test/test.js

```diff
@@ -22,234 +22,667 @@ const awsContext = require('aws-lambda-mock-context');
+      createserviceReturn.catch((err) => {
 
-    /*
-    * Given a failed http Post attempt, handler() indicates there was an error with the Jenkins job
-    * @params {object, object, function} default event, aws context, callback
-    * @returns index.handler() should return a descriptive error message concerning a failed Jenkins Job
-    */
-    it("should indicate an error occured with Jenkins setup if the POST attempt fails", () => {
-      //wrapping the logger messages to console to check for error message
-      stub = sinon.stub(logger, "error", spy);
-      //trigger the spy wrapping the logger by calling handler() with valid params
-      var callFunction = index.handler(event, context, callback);
-      stub.restore();
-      //spy has already been called for previous case, so the arguments passed to the logger.error()
-      //are contained in an array at spy.args[1]
-      var bool = spy.args[1][0].includes("Error while starting Jenkins job: ");
-      assert.isTrue(bool);
+        if (err.message && err.message === errMessage) {
+          bool = true;
+        }
+        assert.isTrue(bool);
+      })
+      reqStub.restore();
     });
+
+  })
+  describe("startServiceOnboarding", () => {
+    let config, service_id, event;
+    beforeEach(() => {
+      event = {
+        "stage": "test",
+        "headers": {
+          "Authorization": "fr1end$hip_1s_mAg1c"
+        },
+        "principalId": "@pp1eJack",
+        "body": {
+          "service_name": "test-service",
+          "service_type": "function",
+          "domain": "test-domain",
+          "runtime": "nodejs",
+          "approvers": ['tw1light_$pArkle'],
+          "rateExpression": "1 * * * ? *",
+          "slack_channel": "mlp_fim",
+          "require_internal_access": false,
+          "create_cloudfront_url": false //, //?
+          //"enableEventSchedule" : false
+        }
+      };
+      service_creation_data =  event.body;
+      config = configObj(event);
+      service_id = "ghd93-3240-2343";
+    })
+    it("should return error message if request to start jenkins job failed ", () => {
+      let bool = false;
+      let responseObject = {
+        statusCode: 401,
+        body: {
+          message: "401 UNAUTHORIZED Jenkins Job Not triggered"
+        }
+      };
+      let reqStub = sinon.stub(request, "Request", (obj) => {
+        let errObject = {
+          message: responseObject.body.message,
+          jenkins_api_failure: true
+        }
+        return obj.callback(errObject, responseObject, responseObject.body);
+      });
+      index.startServiceOnboarding(service_creation_data, config, service_id).then(() => {
+        assert.fail();
+      }).catch((err) => {
+        if (err.jenkins_api_failure && err.message != null) {
+          bool = true;
+        }
+        assert.isTrue(bool);
+      })
+      reqStub.restore();
+    })
+    it("should return error message when jenkins job has failed", () => {
```

Comment:
give extra lines between every 'it'. other place too.

Reply:
done